### PR TITLE
CUST-CKFLIGHTF4 Custom Designed target

### PR DESCRIPTION
**Pull request by ckflight**
I have my custom flight controller design where i have used for many years with my firmware as well as betaflight.

I was using STM32F4DISCOVERY firmware and CLI to program it. However after 4.5.0 update, i could not use STM32F4DISCOVERY. The cli configuration was not enough to set peripherals correctly. 

I used other boards' config files and managed it to work. However i want my own config file now to be added to the betaflight targets if it is possible please.

The flight controller is tested many times and working. I can prove it as well.

The pin definitions are correct and DMA streams are selected accordingly.

Some Board Details:
MCU: STM32F405
Gyro&Acc: ICM20602
Barometer: MS5611
Mag: MLX90393 (it is not supported by betaflight i suppose so i did not add it to pin config)
On board Analog OSD: MAX7456 
On board I2C external pin : I2C2
On board UART external pin: USART1 RX TX
On board UART external pin: UART4 RX TX
Selectable interverted rx receiver for sbus at UART6
2 User Leds
1 RGB (3.3v to 5v converted)
1 Buzzer with 4KHz pwm
SPI SDCard interface
1/10 scaled Lipo and current reading to ADC inputs

I am adding the board images as well.

### CODE REVIEW

**Comment by nerdCopter** [comment]
4.5 and beyond now uses config.h files.
documentation: https://betaflight.com/docs/development/manufacturer/creating-configuration
repository: https://github.com/betaflight/config

**Review by nerdCopter — APPROVED** [review]

**Comment by nerdCopter** [comment]
@ckflight , please see https://betaflight.com/docs/development/Git#recommended-git-global-environment for recommended global settings which will mitigate line-ending issues.

**Comment by ckflight** [comment]
Hi, sorry for this mistake, i dont have experience on developing over github. I just created this file using sublime text and added to my github fork repository and the pulled it. What is the correct way of doing it. It is cross platform new line encoding issue i suppose.

**Comment by nerdCopter** [comment]
such is perfectly fine, i use sublimetext myself. just set your `git` global environment as suggested. may have to save/commit the file again.

**Review by ckflight — COMMENTED** [review]
I edited file with github desktop app and used clrf settings as below so i hope it should work now

[core]
	autocrlf = input
[filter "lfs"]
	clean = git-lfs clean -- %f
	smudge = git-lfs smudge -- %f
	process = git-lfs filter-process
	required = true
[user]
	name = Cenk Keskin
	email = cnkkskn001@gmail.com

**Comment by haslinghuis** [comment]
Should revise the board to not share SPI but as I understand the board will not be public available.

```
set sdcard_spi_bus = 3
set max7456_spi_bus = 3
```

File: configs/default/CUST-CKFLIGHTF4.config [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+set beeper_od = OFF
+set beeper_frequency = 4000
+set system_hse_mhz = 8
+set max7456_spi_bus = 3
+set current_meter = ADC
+set battery_meter = ADC
+set gyro_1_bustype = SPI
+set gyro_1_spibus = 1
+set gyro_1_sensor_align = CW0
+set gyro_1_align_yaw = 0
```
Comment (haslinghuis) [inline configs/default/CUST-CKFLIGHTF4.config:105]: Guess these are the default setting.
```suggestion
```

**Comment by ckflight** [comment]
> Should revise the board to not share SPI but as I understand the board will not be public available.
> 
> ```
> set sdcard_spi_bus = 3
> set max7456_spi_bus = 3
> ```

It is already divided. SD Card uses spi 2 and OSD uses spi 3. I have corrected the mistake on the config file.
I made lots of open source projects for others to learn such as radar, linux board and blhelis esc which are available at my github page. 

In the near future i want to sell flight controller since this hobby is not well developed in Türkiye and i want to contribute with hardwares to the comunity. For now, I just wanted to add the board to betaflight environment since 4.5.0 now requires it. I have H7 design as well if this one goes well, i want to add that desing as well

**Review by nerdCopter — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Comment by haslinghuis** [comment]
Lols, guess we need to head on to config repo instead. It was planned to lockout this repo for further development.
Please head over to https://github.com/betaflight/config

image

**Comment by ckflight** [comment]
> Lols, guess we need to head on to config repo instead. It was planned to lockout this repo for further development. Please head over to https://github.com/betaflight/config
> 
> image

Thank you so much i just moved to that part now. I will create config.h file

**Pull request closed without merging**